trusted-types/trusted-types-eval-reporting-no-unsafe-eval.html

```diff
@@ -59,56 +31,35 @@
 
   const scriptyPolicy = trustedTypes.createPolicy('allowEval', a_policy);
 
-  // Provoke/wait for a CSP violation, in order to be sure that all previous
-  // CSP violations have been delivered.
-  function promise_flush() {
-    return promise_violation("object-src 'none'");
-  }
-  function flush() {
-   expect_throws(_ => {
-      var o = document.createElement('object');
-      o.type = "application/x-shockwave-flash";
-      document.body.appendChild(o);
-    });
-  }
-
   window.script_run_beacon = 'never_overwritten';
 
-  promise_test(t => {
-    let p = Promise.resolve()
-        .then(promise_violation("require-trusted-types-for 'script'"))
-        .then(promise_flush());
-    expect_throws(_ => eval('script_run_beacon="should not run"'));
+  promise_test(async t => {
+    let violation = await trusted_type_violation_for(EvalError, _ =>
```

Comment:
Should we be asserting based on this violation object?

Reply:
I think you are right, probably we need a 

`assert_true(violation.originalPolicy.includes("require-trusted-types-for 'script'"));`

to keep the same checks as the existing test.